---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: corob-msft
ms.author: corob
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 35f138288cc46e1b7a4be34b530f07df7f66caff
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 02/15/2020
ms.locfileid: "77270185"
---
# <a name="c28105"></a>C28105
경고 C28105: 예외로 인해 리소스 누수가 발생 합니다.

 예외가 발생 하면 지정 된 리소스가 해제 되지 않습니다. 경로에서 지정 된 문은 예외를 발생 시킬 수 있습니다. 이 경고는 [C28103](../code-quality/c28103.md)와 유사 합니다. 단,이 경우에는 예외가 수반 됩니다.

## <a name="example"></a>예제
 다음 코드 예제에서는 이 경고를 생성합니다.

```cpp
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 다음 코드 예제에서는 이 경고를 방지합니다.

```cpp
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```
